src/flow/mod.rs

```diff
@@ -295,6 +296,22 @@ impl Blender {
         r
     }
 
+    /// Initiaties log recovery by sending a
+    /// StartRecovery packet to each base node domain.
+    pub fn recover(&mut self) {
+        info!(self.log, "Recovering from log");
+        let indices = self.inputs()
+            .iter()
+            .map(|&(_input, ref node)| node.domain())
+            .collect::<Vec<_>>();
+
+        for index in indices {
+            let domain = self.domains.get_mut(&index).unwrap();
+            domain.send(box payload::Packet::StartRecovery).unwrap();
+            domain.wait_for_ack().unwrap();
```

Comment:
Is it necessary to have domains recover sequentially? Could they all do so in parallel?

Reply:
Yep!